commit 26e8b2896cec152643701ea8d8636ed5d561788b
Author: Brice Goglin <Brice.Goglin@inria.fr>
Date:   Tue Dec 31 19:35:59 2013 +0100

    properly abort xen component instantiate if not running as root in dom0

diff --git a/src/topology-xen.c b/src/topology-xen.c
index b74f9f5..28526fe 100644
--- a/src/topology-xen.c
+++ b/src/topology-xen.c
@@ -305,7 +305,10 @@ hwloc_xen_component_instantiate(struct hwloc_disc_component *component,
 
   /* This will fail if we are not running as root in dom0. */
   priv->xch = xc_interface_open(NULL, NULL, 0);
-  assert(priv->xch && "Are you running as root in dom0?");
+  if (!priv->xch) {
+    hwloc_debug("xc_interface_open() failed, are you running as root in dom0? Disabling xen component.");
+    goto err;
+  }
 
   backend->private_data = priv;
   backend->discover = hwloc_xen_discover;
@@ -314,7 +317,7 @@ hwloc_xen_component_instantiate(struct hwloc_disc_component *component,
   return backend;
 
  err:
-  if (priv->xch)
+  if (priv && priv->xch)
     xc_interface_close(priv->xch);
   free(priv);
   free(backend);

